orca-pipelinetemplate/src/main/java/com/netflix/spinnaker/orca/pipelinetemplate/v1schema/graph/transform/PipelineConfigInheritanceTransform.java

```diff
@@ -53,4 +54,26 @@ public void visitPipelineTemplate(PipelineTemplate pipelineTemplate) {
       pc.setNotifications(Collections.emptyList());
     }
   }
+
+  @Override
+  public void visitPipelineTemplate(V2PipelineTemplate pipelineTemplate) {
```

Comment:
Just so I understand, if we add a new top-level field to regular pipelines, do we need to make a code change here for it to work even if a user doesn't want to inherit it?

Reply:
With the new refactoring commits, top level fields will be honored if specified in the templates but won't be able to be overridden/merged in the configurations without code changes.